### PR TITLE
Multithread forest application to matrices

### PR DESCRIPTION
**Pull request by salbert83**
I guess one could multithread application of a forest to a vector instead.  I think it is better to do it at this level.

### CODE REVIEW

**Comment by ablaom** [comment]
For prediction in a forest I agree that it makes sense to always use multithreading. @OkonSamuel Your thoughts? (And can you please review?)

There are possiblities for multithreading in training both forests and an individual tree, but let's leave that for a separate PR.

File: src/classification/main.jl [hunk 1]
```diff
@@ -271,7 +271,7 @@ end
 function apply_forest(forest::Ensemble{S, T}, features::AbstractMatrix{S}) where {S, T}
     N = size(features,1)
     predictions = Array{T}(undef, N)
-    for i in 1:N
+    Threads.@threads for i in 1:N
```
Comment (OkonSamuel) [inline src/classification/main.jl:274]: I think we should check for the case of `Threads.nthreads() == 1`. For this case due to task overhead single threaded implementation is better.

File: src/classification/main.jl [hunk 2]
```diff
@@ -271,7 +271,7 @@ end
...
         predictions[i] = apply_forest(forest, features[i, :])
```
Comment (OkonSamuel) [inline src/classification/main.jl:275]: Note: With the current implementation, the speed improvements from using multithreading will come with significantly increased memory imprint especially for large forests. We could re-write the existing codebase to reduce this.

**Review by OkonSamuel — CHANGES_REQUESTED** [review]
I think we should check for the case of `Threads.nthreads() == 1`. For this case due to task overhead single threaded implementation is better.
Other than this, LGTM.
I think there is room for improving performance, but this can be addressed later.

File: src/classification/main.jl (hunk 2 above)
Comment (salbert83) [reply src/classification/main.jl:275]: Do these criticisms apply similarly to the current use of multithreading in building a forest?

**Pull request closed without merging**